spec/factories/ems_folder.rb

```diff
@@ -3,7 +3,7 @@
     sequence(:name) { |n| "Test Folder #{seq_padded_for_sorting(n)}" }
   end
 
-  factory :datacenter, :parent => :ems_folder do
-    is_datacenter true
+  factory :datacenter, :parent => :ems_folder, :class => "Datacenter" do
+    type "Datacenter"
```

Comment:
This should not be necessary if you have STI and are using `:class => "Datacenter"`